src/CosmosDB/CosmosDB/CosmosDBAccount/RestoreAzCosmosDBAccount.cs

```diff
@@ -0,0 +1,132 @@
+            {
+                utcRestoreDateTime = RestoreTimestampInUtc;
+            }
+            else
+            {
+                utcRestoreDateTime = RestoreTimestampInUtc.ToUniversalTime();
+            }
+
+            WriteObject("RestoreTimestamp in UTC: " + utcRestoreDateTime);
+
+            List<RestorableDatabaseAccountGetResult> restorableDatabaseAccounts = CosmosDBManagementClient.RestorableDatabaseAccounts.ListWithHttpMessagesAsync().GetAwaiter().GetResult().Body.ToList();
+
+            RestorableDatabaseAccountGetResult sourceAccountToRestore = null;
+            List<RestorableDatabaseAccountGetResult> accountsWithMatchingName = restorableDatabaseAccounts.Where(databaseAccount => databaseAccount.AccountName.Equals(SourceDatabaseAccountName, StringComparison.OrdinalIgnoreCase)).ToList();
+            if (accountsWithMatchingName.Count > 0)
+            {
+                foreach (RestorableDatabaseAccountGetResult restorableAccount in accountsWithMatchingName)
+                {
+                    if (restorableAccount.CreationTime.HasValue &&
+                        restorableAccount.CreationTime < utcRestoreDateTime)
+                    {
+                        if (!restorableAccount.DeletionTime.HasValue || restorableAccount.DeletionTime > utcRestoreDateTime)
+                        {
+                            sourceAccountToRestore = restorableAccount;
+                            break;
+                        }
+                    }
+                }
+            }
+
+            if (sourceAccountToRestore == null)
+            {
+                WriteWarning($"No database accounts found with matching account name {SourceDatabaseAccountName} that was alive at given utc-timestamp {utcRestoreDateTime}");
+                return;
+            }
+
+            DatabaseAccountCreateUpdateProperties databaseAccountCreateUpdateProperties = null;
+            PSRestoreParameters restoreParameters = new PSRestoreParameters()
+            {
+                RestoreSource = sourceAccountToRestore.Id,
+                RestoreTimestampInUtc = utcRestoreDateTime,
+                DatabasesToRestore = DatabasesToRestore
+            };
+
+            Collection<Location> LocationCollection = new Collection<Location>();
+            Location loc = new Location(locationName: Location, failoverPriority: 0);
+            LocationCollection.Add(loc);
+
+            databaseAccountCreateUpdateProperties = new RestoreReqeustDatabaseAccountCreateUpdateProperties()
+            {
+                RestoreParameters = restoreParameters.ToSDKModel(),
+                Locations = LocationCollection
+            };
+
+            DatabaseAccountCreateUpdateParameters databaseAccountCreateUpdateParameters = new DatabaseAccountCreateUpdateParameters(location: sourceAccountToRestore.Location, name: TargetDatabaseAccountName, properties: databaseAccountCreateUpdateProperties)
+            {
+                Kind = "GlobalDocumentDB"
```

Comment:
use the SDK enum. Why cant this be MongoDB ?

Reply:
Fixed it. Checking the api type from source account and setting on the restored account. 
(Actually the kind passed here does not matter. The restored account is always the same kind as source)